src/encoded/static/components/viz/Workflow/Graph.js

```diff
@@ -254,12 +283,35 @@ export default class Graph extends React.Component {
                         nodesInColumn.splice(afterThisIdx + 1, 0, gN);
                     }
                 });
-                // Update own indices (not used for anything currently xcept debugging)
-                _.forEach(nodesInColumn, function(n, index){
-                    n.indexInColumn = index;
-                });
             }
             
+            if (_.every(nodesInColumn, function(n){ return n.type === 'step'; })){
```

Comment:
This part will very likely change in future.

Currently if any step node(s) in column of step nodes has more inputs than node w/ lowest # of inputs in same column: gathers such nodes and injects them into middle of column.

Doesn't do anything smarter than that at the moment. One possibility for the future could be to try to calculate subsequent edge intersections arising from placement at each index and then select placement to minimize it.